### PR TITLE
Add Validation for Email List

### PR DESCRIPTION
**Pull request by codingmusician**
### Tickets
[ET-1621]
[ET-1622]

### Description
The new Tickets Emails feature will have settings fields that allow for a comma separated listing of emails. We needed to create a way to validate each email in the field before saving them. We're calling the `validation_type` `email_list`.

[ET-1621]: https://theeventscalendar.atlassian.net/browse/ET-1621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-1622]: https://theeventscalendar.atlassian.net/browse/ET-1622?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Validate.php [hunk 1]
```diff
@@ -499,6 +499,38 @@ public function email(  ) {
...
+			$emails = preg_split( '[,|;]', $candidate );
+			$this->result->valid = true;
+			$sanitized_emails = [];
+
+			foreach ( $emails as $email ) {
+				if ( ! filter_var( trim( $email ), FILTER_VALIDATE_EMAIL ) ) {
+					$this->result->valid = false;
+					break;
+				}
+				$sanitized_emails[] = filter_var( trim( $email ), FILTER_SANITIZE_EMAIL );
```
Comment (bordoni) [inline src/Tribe/Validate.php:518]: https://developer.wordpress.org/reference/functions/sanitize_email/

File: src/Tribe/Validate.php [hunk 2]
```diff
@@ -499,6 +499,38 @@ public function email(  ) {
...
+		 * @since TBD
+		 */
+		public function email_list() {
+			$candidate = trim( $this->value );
+			$emails = preg_split( '[,|;]', $candidate );
+			$this->result->valid = true;
+			$sanitized_emails = [];
+
+			foreach ( $emails as $email ) {
+				if ( ! filter_var( trim( $email ), FILTER_VALIDATE_EMAIL ) ) {
```
Comment (bordoni) [inline src/Tribe/Validate.php:514]: avoid doing it twice, just save on a variable.

File: src/Tribe/Validate.php [hunk 3]
```diff
@@ -499,6 +499,38 @@ public function email(  ) {
...
 		}
 
+		/**
+		 * Validates and sanitizes a list of email addresses.
+		 *
+		 * @since TBD
+		 */
+		public function email_list() {
+			$candidate = trim( $this->value );
+			$emails = preg_split( '[,|;]', $candidate );
```
Comment (bordoni) [inline src/Tribe/Validate.php:509]: I think we should break this method down a bit further.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Validate.php (hunk 3 above)
Comment (codingmusician) [reply src/Tribe/Validate.php]: Made a few tweaks after doing a little more testing with it. Added some more comments, too. What did you have in mind as far as breaking it down?

Comment (codingmusician) [reply src/Tribe/Validate.php]: Pinging you again on this @bordoni

File: src/Tribe/Validate.php [hunk 4]
```diff
@@ -499,6 +499,46 @@ public function email(  ) {
...
+		 *
+		 * @since TBD
+		 */
+		public function email_list() {
+			$value = trim( $this->value );
+			$this->result->valid = true;
+			$sanitized_emails = [];
+
+			// Break emails into an array.
+			$emails = preg_split( '[,|;]', $value );
```
Comment (rafsuntaskin) [inline src/Tribe/Validate.php:513]:
```suggestion
			$emails = preg_split( '/[,;]+/', $value );
```

Comment (rafsuntaskin) [reply src/Tribe/Validate.php:513]: Same result but more specific.

File: src/Tribe/Validate.php (hunk 3 above)
Comment (codingmusician) [reply src/Tribe/Validate.php]: One more try @bordoni ?

**Review by juanfra — APPROVED** [review]

**Pull request merged**